src/mame/machine/decocass.cpp

```diff
@@ -1224,274 +1281,301 @@ void decocass_state::machine_reset()
 }
 
-MACHINE_RESET_MEMBER(decocass_state,ctsttape)
+void decocass_type1_state::machine_start()
+{
+	save_item(NAME(m_type1_inmap));
+	save_item(NAME(m_type1_outmap));
+}
+
+void decocass_type1_state::machine_reset()
+{
+	decocass_state::machine_reset();
+
+	m_dongle_r = read8_delegate(FUNC(decocass_type1_state::decocass_type1_r),this);
+	m_type1_inmap = MAKE_MAP(0,1,2,3,4,5,6,7);
+	m_type1_outmap = MAKE_MAP(0,1,2,3,4,5,6,7);
+}
+
+void decocass_type2_state::machine_start()
+{
+	save_item(NAME(m_type2_d2_latch));
+	save_item(NAME(m_type2_xx_latch));
+	save_item(NAME(m_type2_promaddr));
+}
+
+void decocass_type2_state::machine_reset()
 {
 	decocass_state::machine_reset();
+
+	LOG(0,("dongle type #2 (CS82-007)\n"));
+	m_dongle_r = read8_delegate(FUNC(decocass_type2_state::decocass_type2_r),this);
+	m_dongle_w = write8_delegate(FUNC(decocass_type2_state::decocass_type2_w),this);
+
+	m_type2_d2_latch = 0;
+	m_type2_xx_latch = 0;
+	m_type2_promaddr = 0;
+}
+
+void decocass_type3_state::machine_start()
+{
+	save_item(NAME(m_type3_ctrs));
+	save_item(NAME(m_type3_d0_latch));
+	save_item(NAME(m_type3_pal_19));
+	save_item(NAME(m_type3_swap));
+}
+
+void decocass_type3_state::machine_reset()
+{
+	decocass_state::machine_reset();
+
+	m_dongle_r = read8_delegate(FUNC(decocass_type3_state::decocass_type3_r),this);
+	m_dongle_w = write8_delegate(FUNC(decocass_type3_state::decocass_type3_w),this);
+
+	m_type3_ctrs = 0;
+	m_type3_d0_latch = 0;
+	m_type3_pal_19 = 0;
+	m_type3_swap = 0;
+}
+
+void decocass_type4_state::machine_start()
+{
+	save_item(NAME(m_type4_ctrs));
+	save_item(NAME(m_type4_latch));
+}
+
+void decocass_type4_state::machine_reset()
+{
+	decocass_state::machine_reset();
+
+	LOG(0,("dongle type #4 (32K ROM)\n"));
+	m_dongle_r = read8_delegate(FUNC(decocass_type4_state::decocass_type4_r),this);
+	m_dongle_w = write8_delegate(FUNC(decocass_type4_state::decocass_type4_w),this);
+
+	m_type4_ctrs = 0;
+	m_type4_latch = 0;
+}
+
+void decocass_type5_state::machine_start()
+{
+	save_item(NAME(m_type5_latch));
+}
+
+void decocass_type5_state::machine_reset()
+{
+	decocass_state::machine_reset();
+
+	LOG(0,("dongle type #5 (NOP)\n"));
+	m_dongle_r = read8_delegate(FUNC(decocass_type5_state::decocass_type5_r),this);
+	m_dongle_w = write8_delegate(FUNC(decocass_type5_state::decocass_type5_w),this);
+
+	m_type5_latch = 0;
+}
+
+void decocass_nodong_state::machine_reset()
+{
+	decocass_state::machine_reset();
+	LOG(0, ("no dongle\n"));
+	m_dongle_r = read8_delegate(FUNC(decocass_nodong_state::decocass_nodong_r), this);
+}
+
+MACHINE_RESET_MEMBER(decocass_type1_state,ctsttape)
+{
+	decocass_type1_state::machine_reset();
 	LOG(0,("dongle type #1 (DE-0061)\n"));
-	m_dongle_r = read8_delegate(FUNC(decocass_state::decocass_type1_r),this);
 	m_type1_map = type1_pass_136_table;
 }
 
-MACHINE_RESET_MEMBER(decocass_state,chwy)
+MACHINE_RESET_MEMBER(decocass_type1_state,chwy)
 {
-	decocass_state::machine_reset();
+	decocass_type1_state::machine_reset();
```

Comment:
When calling a member function of your own class, you don't need to qualify it like this - you should be able to just call `machine_reset()` in all these cases.  It's less visual clutter.